src/_pytest/config/__init__.py

```diff
@@ -457,18 +461,18 @@ def parse_hookimpl_opts(self, plugin: _PluggyPlugin, name: str):
         method = getattr(plugin, name)
         # Consider only actual functions for hooks (#3775).
         if not inspect.isroutine(method):
-            return
+            return None
         # Collect unmarked hooks as long as they have the `pytest_' prefix.
-        return _get_legacy_hook_marks(
+        return _get_legacy_hook_marks(  # type: ignore[return-value]
```

Comment:
This can be improved, but the legacy hook stuff is going to be removed in pytest 8.1 so I didn't bother.